numpy/__init__.pxd

```diff
@@ -500,6 +500,12 @@ cdef extern from "numpy/arrayobject.h":
     void* PyArray_MultiIter_DATA(broadcast multi, npy_intp i) nogil
     void PyArray_MultiIter_NEXTi(broadcast multi, npy_intp i) nogil
     bint PyArray_MultiIter_NOTDONE(broadcast multi) nogil
+    npy_intp PyArray_MultiIter_SIZE(broadcast multi) nogil
+    int PyArray_MultiIter_NDIM(broadcast multi) nogil
+    npy_intp PyArray_MultiIter_INDEX(broadcast multi) nogil
+    int PyArray_MultiIter_NUMITER(broadcast multi) nogil
+    npy_intp* PyArray_MultiIter_DIMS(broadcast multi) nogil
+    void** PyArray_MultiIter_ITERS(broadcast multi) nogil
```

Comment:
@mattip do you have an opinion about adding Macros?  We could make it static inline functions, I guess.

Adding the structs would maybe make following up with making the struct opaque easier.  Although, unless we change the layout now we would need to wait for a 3.0 "please recompile" release again.
(Unless we make some of these proper functions in the function table.)

(TBH, I am not sure if it is worth it, although the structs are limited to 32 dims and 32 iterators for not very good reasons probably.)

Reply:
I don't think we should touch the cython 0.29.x file, and at some point would prefer we deprecate cython<3.0 use.

Reply:
Reverted in [`fe86e38` (#24658)](https://github.com/numpy/numpy/pull/24658/commits/fe86e382757c781bb7e0ae5eeb51558e9102b394)